HackerRank - Binary Calculator/js/binaryCalculator.js

```diff
@@ -0,0 +1,86 @@
+let button1 = document.getElementById("btn1");
+let button0 = document.getElementById("btn0");
+
+let buttonClear = document.getElementById("btnClr");
+
+let buttonSum = document.getElementById("btnSum");
+let buttonSub = document.getElementById("btnSub");
+let buttonMulti = document.getElementById("btnMul");
+let buttonDiv = document.getElementById("btnDiv");
+
+
+let buttonEqual = document.getElementById("btnEql");
+let resultDisplay = document.getElementById("res");
+
+let number = ""
+let operator = ""
+let operand1 = ""
+let operand2 = ""
+let result = ""
+
+let bolconditional = true
+
+function writeNumber (element) {
+    number = document.getElementById(element.target.id).innerHTML
+    if (bolconditional) {
+        resultDisplay.innerHTML += number
+    } else {
+        operand2 += number
+        resultDisplay.innerHTML = operand1 + operator + operand2
+        
+}};
+
+function writeOperator (element) {
+    operator = document.getElementById(element.target.id).innerHTML
+    bolconditional = false
+    
+    if (operand1 === ""){
+        operand1 = resultDisplay.innerHTML
+        resultDisplay.innerHTML = operand1 + operator
+    } else if(operand2 != ""){
+        equal()
+        operand1 = result
+    } else 
+    resultDisplay.innerHTML = operand1 + operator 
+};
+
+function clearResult () {
+    resultDisplay.innerHTML = ""
```

Comment:
You could create a data object here for the values in the result. E.g. In the beginning of the file you can create:

```javascript
let resultDefault = {'operand1': 0, 'operand2': 0, 'numer': "", ...}
let result = resultDefault

// And 
function clearResult () {
   result = resultDefault
  return 

}

Reply:
You can also separate the default in a JSON file.